src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -153,71 +153,56 @@
+        this.frame = panel.frame();
         this.panel = panel;
-        this.entry = Objects.requireNonNull(entry);
-        entry.registerListener(this);
-        entryType = EntryTypes.getTypeOrDefault(entry.getType(),
-                this.frame.getCurrentBasePanel().getBibDatabaseContext().getMode());
-
-        displayedBibEntryType = entry.getType();
 
         writeXmp = new WriteXMPEntryEditorAction(panel, this);
 
-        BorderLayout borderLayout = new BorderLayout();
-        setLayout(borderLayout);
-        setupToolBar();
+        layout = new BorderLayout();
+        setLayout(layout);
 
         container = OS.LINUX ? new CustomJFXPanel() : new JFXPanel();
+        // Create type-label
+        typeLabel = new TypeLabel("");
+        setEntry(entry);
+        setupToolBar();
+        container.setScene(new Scene(tabbed));
+        add(container, BorderLayout.CENTER);
 
-        container.addKeyListener(new KeyAdapter() {
+        DefaultTaskExecutor.runInJavaFXThread(() -> {
+                    EasyBind.subscribe(tabbed.getSelectionModel().selectedItemProperty(), tab -> {
+                        EntryEditorTab activeTab = (EntryEditorTab) tab;
+                        if (activeTab != null) {
+                            activeTab.notifyAboutFocus();
+                        }
+                    });
+                });
 
-            @Override
-            public void keyPressed(java.awt.event.KeyEvent e) {
+        setupKeyBindings();
+    }
+
+    public void setEntry(BibEntry entry) {
+        this.entry = Objects.requireNonNull(entry);
+        entry.registerListener(this);
+        entryType = EntryTypes.getTypeOrDefault(entry.getType(),
+                this.frame.getCurrentBasePanel().getBibDatabaseContext().getMode());
+
+        displayedBibEntryType = entry.getType();
 
-                //We need to consume this event here to prevent the propgation of keybinding events back to the JFrame
-                Optional<KeyBinding> keyBinding = Globals.getKeyPrefs().mapToKeyBinding(e);
-                if (keyBinding.isPresent()) {
-                    switch (keyBinding.get()) {
-                        case CUT:
-                        case COPY:
-                        case PASTE:
-                        case CLOSE_ENTRY_EDITOR:
-                        case DELETE_ENTRY:
-                        case SELECT_ALL:
-                        case ENTRY_EDITOR_NEXT_PANEL:
-                        case ENTRY_EDITOR_NEXT_PANEL_2:
-                        case ENTRY_EDITOR_PREVIOUS_PANEL:
-                        case ENTRY_EDITOR_PREVIOUS_PANEL_2:
-                            e.consume();
-                            break;
-                        default:
-                            //do nothing
-                    }
-                }
-            }
-        });
         DefaultTaskExecutor.runInJavaFXThread(() -> {
-            addTabs(lastTabName);
+            addTabs(this.getVisibleTabName());
```

Comment:
is `getVisibleTabName` used somewhere outside of this class? if not then it can be converted to just a field and don't need to be exposed.

Reply:
It is used in the `BasePanel` and the main table, so it needs to be visible.